PixiEditor/Models/Tools/ToolSettings/Toolbars/Toolbar.cs

```diff
@@ -20,26 +21,46 @@ public virtual Setting GetSetting(string name)
             return Settings.FirstOrDefault(x => string.Equals(x.Name, name, StringComparison.CurrentCultureIgnoreCase));
         }
 
+        /// <summary>
+        ///     Gets setting with given type T in toolbar by name.
+        /// </summary>
+        /// <param name="name">Setting name, non case sensitive</param>
+        /// <returns></returns>
+        public Setting<T> GetSetting<T>(string name)
+        {
+            Setting setting =  Settings.FirstOrDefault(currentSetting => string.Equals(currentSetting.Name, name, StringComparison.CurrentCultureIgnoreCase));
+
+            if (setting == null || !(setting is Setting<T> convertedSetting))
+                return null;
+
+            return convertedSetting;
+        }
```

Comment:
In my opinion would be better to insert in `T` the `Setting` based class, not the `Value` type
So instead of `Setting<T>` it would return `T` Where `T` is a class that inherits from `Setting`. I described this more detailed here https://github.com/PixiEditor/PixiEditor/pull/51#issuecomment-695974432